editor/src/document/properties_panel_message_handler.rs

```diff
@@ -0,0 +1,379 @@
+use super::layer_panel::LayerDataTypeDiscriminant;
+use crate::document::properties_panel_message::TransformOp;
+use crate::layout::layout_message::LayoutTarget;
+use crate::layout::widgets::{
+	IconLabel, LayoutRow, NumberInput, PopoverButton, Separator, SeparatorDirection, SeparatorType, TextInput, TextLabel, Widget, WidgetCallback, WidgetHolder, WidgetLayout,
+};
+use crate::message_prelude::*;
+
+use graphene::document::Document as GrapheneDocument;
+use graphene::layers::layer_info::{Layer, LayerDataType};
+use graphene::{LayerId, Operation};
+
+use glam::{DAffine2, DVec2};
+use serde::{Deserialize, Serialize};
+use std::f64::consts::PI;
+
+trait DAffine2Utils {
+	fn width(&self) -> f64;
+	fn update_width(self, new_width: f64) -> Self;
+	fn height(&self) -> f64;
+	fn update_height(self, new_height: f64) -> Self;
+	fn x(&self) -> f64;
+	fn update_x(self, new_x: f64) -> Self;
+	fn y(&self) -> f64;
+	fn update_y(self, new_y: f64) -> Self;
+	fn rotation(&self) -> f64;
+	fn update_rotation(self, new_rotation: f64) -> Self;
+}
```

Comment:
Could you potentially split this into Mat2Utils and DAffine2 Utils?
We don't have to do this pre launch though

Reply:
I think most of this will end up being removed anyways by @0HyperCube when he updates this code to properly use the bounding boxes